src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ManagedSslBio.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        internal static class ManagedSslBio
+        {
+            private unsafe static ReadDelegate s_readDelegate;
+            private unsafe static WriteDelegate s_writeDelegate;
+
+            static ManagedSslBio() => Initialize();
+
+            internal static SafeBioHandle CreateManagedSslBio() => Crypto.CreateManagedSslBio();
+            
+            private unsafe static void Initialize()
+            {
+                s_writeDelegate = Write;
+                s_readDelegate = Read;
+                Crypto.InitManagedSslBioMethod(s_writeDelegate, s_readDelegate);
+            }
+
+            internal static void BioSetGCHandle(SafeBioHandle bio, GCHandle handle)
+            {
+                IntPtr pointer;
+                if (handle.IsAllocated)
+                {
+                    pointer = GCHandle.ToIntPtr(handle);
+                }
+                else
+                {
+                    pointer = IntPtr.Zero;
+                }
+
+                Crypto.BioSetAppData(bio, pointer);
+            }
+
+            private static unsafe int Write(IntPtr bio, void* input, int size, IntPtr data)
+            {
+                GCHandle handle = GCHandle.FromIntPtr(data);
+
+                if (handle.IsAllocated && handle.Target is SafeSslHandle.WriteBioBuffer buffer)
```

Comment:
Not sure it ever should, unless someone uses the buffer handle after the managed object is disposed. Then it might be, can remove if you want